Fix nodejs breakage and re-add bear tests

jayvdb opened the pull request.

fneu committed:
.travis.yml: Run coala-bears tests

Reverts 05fdfb72

jayvdb commented:
Amend or cherry-pick, as useful.

jayvdb commented on Dockerfile:
```diff
@@ -135,14 +148,12 @@ RUN luarocks install luacheck
   csslint \
   dockerfile_lint \
-  eslint \
   jshint \
```
Should also remove jshint, if it is installed from openSUSE pkg

jayvdb commented on Dockerfile:
```diff
@@ -135,14 +148,12 @@ RUN luarocks install luacheck
 # NPM setup
 # FIXME: we should use package.json from coala
 RUN npm install -g \
-  alex \
```
I dont know why alex doeent pass. Actually only one of many tests fails :/

AbdealiLoKo replied:
@jayvdb Are you aware of the conflicting package name in the haskell compiler (ghc i think) called alex ? We've had issues before where the binary used is the wrong binary ...

jayvdb replied:
Ya, I did see that, but it is only `AlexBearTest.test_invalid_files` that fails, and I doubt that `alex` is installed on this rather empty openSUSE image.  And https://github.com/coala/coala-bears/commit/fa70e924 added checks for this , and those tests pass.
A neater solution would be to disable only the one failing test.

AbdealiLoKo commented:
Question: Can we just use cib ? Is it not yet stable enough ?
CC @Adrianzatreanu

jayvdb commented:
Circle node is v5.5.0 ; maybe we need to set node to v5 on openSUSE.

jayvdb committed:
Explicitly install nodejs from Leap

Fixes https://github.com/coala/docker-coala-base/issues/40

jayvdb committed:
Extract dependencies from coala-bears package.json

The dependencies were not constrained, which allowed
uncompatible versions of eslint and alex to be installed.
Using the coala-bears package.json ensures the Docker
image has the same tested versions as the CircleCI tests
run regularly for coala-bears.

Also this allows the Docker image to include other dependencies
added to coala-bears when new bears are added.

jayvdb committed:
Remove broken verilator

The openSUSE RPM is broken.
This has been reported upstream.
Temporarily workaround the problem by not including it
in the Docker image.

Fixes https://github.com/coala/docker-coala-base/issues/39

underyx commented on Dockerfile:
```diff
@@ -133,23 +137,10 @@ RUN julia -e 'Pkg.add("Lint")'
+# Extract dependencies from coala-bear package.json
+# typescript is a peer dependency
+RUN npm install -g typescript \
+    $(sed -ne '/~/{s/^[^"]*"//;s/".*"~/@/;s/",*//;p}' coala-bears/package.json)
```
😳 

How reliable is this? Couldn't we just do

```
WORKDIR /coala-bears/package.json
RUN npm install --global
```

or something similar?

jayvdb replied:
No.  That installs `coala-bears` globally, and none of the dependencies are global.

There are a few alternatives, all ugly.

> How reliable is this?

We now have bear testing again, so we can see if it breaks.
See bottom of [log](https://api.travis-ci.org/jobs/179452542/log.txt?deansi=true)

Mixih commented:
ack 
0e44724
000859c
785538c
3584d5d

underyx commented:
unack https://github.com/coala/docker-coala-base/pull/44/commits/000859cd0c3e85316ba932813d2f9f4bb3ee0982 until my comment is answered

feel free to reack it if you've seen my comment and think it's okay, @jayvdb

jayvdb commented:
reack 000859c
